cmake/simdjson-config.cmake.in

```diff
@@ -0,0 +1,9 @@
+# Get the directory containing this file.
+get_filename_component(@PROJECT_NAME@_CURRENT_CONFIG_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+
+include(CMakeFindDependencyMacro)
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+find_dependency(Threads)
+
+# Import targets.
+include("${@PROJECT_NAME@_CURRENT_CONFIG_DIR}/@PROJECT_NAME@-targets.cmake")
```

Comment:
```suggestion
@PACKAGE_INIT@

include(CMakeFindDependencyMacro)
if(@SIMDJSON_ENABLE_THREADS@)
    set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
    find_dependency(Threads)
endif()
# Import targets.
include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake")
```

if consumers may want to know the value of `SIMDJSON_ENABLE_THREADS` you might also want to expose the variable for downstream users by supplying it in the config.
 `CMAKE_CURRENT_LIST_DIR` is ok here because your target files will be installed into the same folder as the config.

Reply:
you might also want to add `check_required_components(<PackageName>)` because it sets `PackageName_FOUND` according to the docs but you need to drop `NO_CHECK_REQUIRED_COMPONENTS_MACRO` in `configure_package_config_file`